src/utils/prototype-for.js

```diff
@@ -0,0 +1,146 @@
+import MicroState from '../microstate';
+
+const { getOwnPropertyNames, getOwnPropertyDescriptor, defineProperty } = Object;
+
+export default function prototypeFor(prototype, attrs) {
```

Comment:
I think the way to avoid having a hidden property is to make `prototypeFor` itself a recursive function that takes a context object, and that every time you recurse into a nested substate, you push that substate onto the context.

Basically, each substate will serve as the prototype for the nested substate's prototype :)